tests/cover/test_regex.py

```diff
@@ -0,0 +1,123 @@
+    ('longNegativeRanges', '^[^a-zA-Z]$'),
+    ('notLiteral', '^[^a][^b][^c]$'),
+    ('shortNegativeRanges', '^[^a-b][^D-E][^7-8]$'),
+    ('negativeSets', '^[^acsG4]$'),
+    ('setsWithHyphens', '^[a\-d][-as][bs-]$'),
+    ('negativeSetsWithHyphens', '^[^a\-d][^-as][^bs-]$'),
+    ('categoryWord', '^\w[^\w]\W$'),
+    ('categoryWhitespace', '^\s[^\s]\S$'),
+    ('categoryDigits', '^\d[^\d]\D$'),
+    ('wordBoundaries', r'\bfoo\b'),
+    ('backRef', '(?P<a>[abc]v[2q])(?P=a)'),
+    ('noncapturingGroup', '^(?:abc)def$'),
+    ('any', '.'),
+    ('positiveLookahead', 'A(?=BCD)'),
+    ('negativeLookahead', 'AB(?!CD)'),
+    #    ("positiveLookbehind","
+    #    ("negativeLookbehind","
+    ('groupRef', '(a|b) \1')
+])
+def test_coverage(name, pattern):
+    """This test gives coverage for the regex generator."""
+    find(st.regex(pattern), lambda x: re.match(pattern, x) is not None)
+
+
+@pytest.mark.parametrize('name,pattern', [
+    ('backslash1', '\\\\'),
+    ('backslash2', r"\\"),
+    ('literal', r"last"),
+    ('lineend', r"foo.$"),
+    ('repetition1', r"ab*"),
+    ('repetition2', r"ab+"),
+    ('repetition3', r"ab?"),
+    ('non-greedy1', r"<.*?>"),
+    ('repetition4', r"a{6}"),
+    ('repetition5', r"a{3,5}"),
+    ('non-greedy2', r"a{3,5}?"),
+    ('escape1', r"\*"),
+    ('escape2', r"\?"),
+    ('set1', r"[amk]"),
+    ('set2', r"[a-z]"),
+    ('set3', r"[0-5][0-9]"),
+    ('set4', r"[0-9A-Fa-f]"),
+    ('escape3', r"[a\-z]"),
+    ('nonrange', r"[a-]"),
+    ('specialset', r"[(+*)]"),
+    ('class1', r"[\w]"),
+    ('class2', r"[\s]"),
+    ('complement1', r"[^5]"),
+    ('complement2', r"[^^]"),
+    ('parenthesis1', r"[()[\]{}]"),
+    ('parenthesis2', r"[]()[{}]"),
+    ('branch1', r"A|B"),
+    ('backref', r"""(?P<quote>['"]).*?(?P=quote)"""),
+    ('positiveLookahead', r"Isaac (?=Asimov)"),
+    ('negativeLookahead', r"Isaac (?!Asimov)"),
+    #    ("positiveLookbehind", r"(?<=abc)def"),
+    #    ("positiveLookbehind2", r"'(?<=-)\w+'"),
+    #    ("groupExistance", r"(<)?(\w+@\w+(?:\.\w+)+)(?(1)>)"),
+    ('groupRef', '(.+) \1'),
+    ('wordBeginning', r'\bfoo\b'),
+    #    ("notWordBeginning", r"py\B"),
+    ('pythonFunction', r'def\s+([a-zA-Z_][a-zA-Z_0-9]*)\s*\(\s*\):'),
+    ('twoWords', r"(\w+) (\w+)"),
+    ('fullName', r"(?P<first_name>\w+) (?P<last_name>\w+)"),
+    ('float', r"(\d+)\.(\d+)"),
+    ('poker', r"^[a2-9tjqk]{5}$"),
+    ('pokerPair', r".*(.).*\1"),
+    ('scanf1', r"."),
+    ('scanf2', r".{5}"),
+    ('scanf3', r"[-+]?\d+"),
+    ('scanf4', r"[-+]?(\d+(\.\d*)?|\.\d+)([eE][-+]?\d+)?"),
+    ('scanf5', r"[-+]?(0[xX][\dA-Fa-f]+|0[0-7]*|\d+)"),
+    ('scanf6', r"[-+]?[0-7]+"),
+    ('scanf7', r"\S+"),
+    ('scanf8', r"\d+"),
+    ('scanf9', r"[-+]?(0[xX])?[\dA-Fa-f]+")
+])
+def test_re_examples(name, pattern):
+    """This test exercises all examples from the python documentation."""
+    find(st.regex(pattern), lambda x: re.match(pattern, x) is not None)
```

Comment:
So currently this test is just asserting that you can find *one* value matching the pattern, when it should really be asserting that all generated values match the pattern.

I think this would be better with a nested given. Something like the following:

```python
@given(st.regex(pattern))
def test_all_generated_data_matches(x):
    assert re.match(pattern, x) is not None
test_all_generated_data_matches()
```

(you need to invoke nested givens explicitly because the test runner won't pick them up)